### ISSUE TITLE
nil error on test single function

### ISSUE THREAD

**Issue by zankich** [nvim-neotest/neotest-go#47]
Hi!

When I run `require('neotest').run.run()` on a single function with a failing test I get the following stacktrace:
```
2023-05-08T14:38:41 Neotest ﭧ WARN neotest-go: ...nkich/.local/share/nvim/plugged/neotest/lua/nio/init.lua:105: The coroutine failed with this message:
...al/share/nvim/plugged/neotest-go/lua/neotest-go/init.lua:223: attempt to index a nil value
stack traceback:
        ...al/share/nvim/plugged/neotest-go/lua/neotest-go/init.lua: in function 'results'
        ...share/nvim/plugged/neotest/lua/neotest/client/runner.lua:131: in function '_run_spec'
        ...share/nvim/plugged/neotest/lua/neotest/client/runner.lua:89: in function <...share/nvim/plugged/neotest/lua/neotest/client/runner.lua:88>
```
Reverting back to [2148ad794e7a5c30c7385a3281f4be91e6b113c4](https://github.com/nvim-neotest/neotest-go/commit/2148ad794e7a5c30c7385a3281f4be91e6b113c4) and executing against the same function I get the proper failed test output.

I was able to reproduce the error with a simple testcase

```
package main

import "testing"

func TestStackTrace(t *testing.T) {
        t.FailNow()
}
```

Testing that function with `b6dc0b1c49569273d863d7b96b6c81b3fc153e82` will throw a stacktrace while testing with `2148ad794e7a5c30c7385a3281f4be91e6b113c4` correctly reports the error.

**Comment by sergii4** [comment]
Hi @zankich, I could reproduce it only once from 10 attempts. Do you have clean steps to reproduce?

**Comment by Snikimonkd** [comment]
i have the same problem on every call of `require("neotest").run.run()`, i don't no how to check plugin's commit, but i tried to update it with `:PackerSync`, so i can assume that this problem reproduces with the latest versions of `neotest` and `neotest-go`

**Comment by zankich** [comment]
I got a PR which fixed the issues for me https://github.com/nvim-neotest/neotest-go/pull/49

**Comment by sergii4** [comment]
Hi @zankich, I believe @michaldziurowski fixed issue in #48. Could you please update plugin and test if it works on your side?

**Comment by tauraamui** [comment]
How can I update the neotest-go runner specifically?
